### ISSUE TITLE
ignoreTrailingComments option for max-len not being respected

### ISSUE THREAD

**Issue by ksmithut** [eslint/eslint#5563]
**What version of ESLint are you using?** 2.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
module.exports = {
  rules: {
    'max-len': [ 2, {
      code: 80,
      tabWidth: 2,
      ignoreUrls: true,
      ignoreTrailingComments: true
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
'use strict'

module.exports = {
  env: 'NODE_ENV' // This sets the config.get('env') to whatever NODE_ENV is set to
}
```

**What did you expect to happen?**

I expected there to not be an eslint error because I've set `ignoreTrailingComments` to `true`, which from the docs means:

> Only ignores comments that are trailing source.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reported an error. It seems to not be respecting the `ignoreTrailingComments` rule

```
  4:1  error  Line 4 exceeds the maximum line length of 80  max-len
```

**Comment by faceleg** [comment]
Confirmed with above example and v2.4.0

**Comment by pvamshi** [comment]
I am trying to fix this. This is my first contribution, so I won't mind if anyone fixes sooner than me . The issue is on this line ( max-len.js, line : 121 ): 
`            comments = ignoreComments || maxCommentLength ? context.getAllComments() : [],
`

 `ignoreComments` = false and `maxCommentLength` = undefined , So `comments` is empty array and stripTrailingComments is not performed. I am trying to figure out what is the correct way to fix it.

**Comment by pvamshi** [comment]
Created pull request #5585
